### PR TITLE
Python: Removes the _async suffix where is not needed

### LINKED ISSUE #4629: Python: Delete all *_async where *_sync doesn't exist
This issue is to ensure the Python SDK is idiomatic with Python. Where there aren't different async/sync methods, the _async suffix should be removed.

### PR DESCRIPTION
**Pull request by juliomenendez**
### Motivation and Context
Delete all *_async where *_sync doesn't exist to make the library more Pythonic.

Fixes #4629 

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: python/samples/kernel-syntax-examples/bing_search_plugin.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ async def main():
     web_plugin = kernel.import_plugin(WebSearchEnginePlugin(connector), "WebSearch")
 
     prompt = "Who is Leonardo DiCaprio's current girlfriend?"
-    search_async = web_plugin["searchAsync"]
-    result = await search_async.invoke_async(prompt)
+    search = web_plugin["searchAsync"]
+    result = await search.invoke_async(prompt)
```
Comment (eavanvalkenburg) [inline python/samples/kernel-syntax-examples/bing_search_plugin.py:30]: not `invoke(`?

File: python/samples/kernel-syntax-examples/google_search_plugin.py [hunk 2]
```diff
@@ -38,10 +38,10 @@ async def main():
 
     # The search query
     prompt = "Who is Leonardo DiCaprio's current girlfriend?"
-    search_async = web_plugin["searchAsync"]
+    search = web_plugin["searchAsync"]
 
     # By default, only one search result is provided
-    result = await search_async.invoke_async(prompt)
+    result = await search.invoke_async(prompt)
```
Comment (eavanvalkenburg) [inline python/samples/kernel-syntax-examples/google_search_plugin.py:44]: same here

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
quick look, overall looking pretty good, some small notes. And we should also add a note to DEV_SETUP.md that a developer should assume everything is async!

File: python/samples/kernel-syntax-examples/bing_search_plugin.py (hunk 1 above)
Comment (juliomenendez) [reply python/samples/kernel-syntax-examples/bing_search_plugin.py:30]: Functions did retain the `invoke_async` and `invoke` methods. So I didn't change those usages.

**Review by moonbox3 — APPROVED** [review]
LGTM

**Review by eavanvalkenburg — APPROVED** [review]
LGTM

**Comment by moonbox3** [comment]
There are some integration tests that still call `kernel.run_async()`, for example `test_azure_oai_chat_service`: the tests should just be `kernel.run()`.

`summary = await retry(lambda: kernel.run(tldr_function, input_str=text_to_summarize))`

**Review by moonbox3 — APPROVED** [review]

**Review by moonbox3 — APPROVED** [review]

**Pull request merged**